### PR TITLE
Fix some clang warnings and general cleanups

### PR DESCRIPTION
**Pull request by frdeso**

### CODE REVIEW

File: lib/graph/query-executor.c [hunk 1]
```diff
@@ -134,7 +134,7 @@ enum bt_query_executor_status bt_query_executor_query(
 	exec_status = (int) status;
 	if (query_exec->canceled) {
 		BT_OBJECT_PUT_REF_AND_RESET(*user_result);
-		status = BT_QUERY_EXECUTOR_STATUS_CANCELED;
+		exec_status = BT_QUERY_EXECUTOR_STATUS_CANCELED;
```
Comment (eepp) [inline lib/graph/query-executor.c:137]: Good catch!

File: plugins/ctf/common/metadata/visitor-generate-ir.c [hunk 2]
```diff
@@ -3484,7 +3484,6 @@ int visit_event_decl(struct ctx *ctx, struct ctf_node *node)
 			stream_class->id = stream_id;
 			g_ptr_array_add(ctx->ctf_tc->stream_classes,
 				stream_class);
-			stream_class = stream_class;
```
Comment (eepp) [inline plugins/ctf/common/metadata/visitor-generate-ir.c:3487]: Wow... :100:

**Comment by eepp** [comment]
Almost okay:

1. Can you rebase on the tip of my branch? The only thing to change should be something about `BT_MESSAGE_ITERATOR_STATUS_CANCELED` in `muxer.c`.
2. 8c6987c02ce1baf06d38f7193ecd91e80ce29def: please name the referred patch instead of using the commit ID because it could change in the future as this whole branch is not merged into `master` yet.
3. 14db54fc2575df34fa021bbf093e8a5ef558b3e5: commit message: put `()` after function name.
4. a452d8c91695cace892280cdd34598bbd0248902: commit message: prefix with `ctf plugin:`.

**Review by eepp — CHANGES_REQUESTED** [review]
Almost okay:

1. Can you rebase on the tip of my branch? The only thing to change should be something about `BT_MESSAGE_ITERATOR_STATUS_CANCELED` in `muxer.c`.
2. 8c6987c02ce1baf06d38f7193ecd91e80ce29def: please name the referred patch instead of using the commit ID because it could change in the future as this whole branch is not merged into `master` yet.
3. 14db54fc2575df34fa021bbf093e8a5ef558b3e5: commit message: put `()` after function name.
4. a452d8c91695cace892280cdd34598bbd0248902: commit message: use `Fix: ctf plugin:` prefix.

**Comment by frdeso** [comment]
I made the requested changes.

**Pull request merged**